woocommerce/woocommerce-android
# Issue/10784 blaze analytics part iii

Closes: #10784


### Description

Final PR for Blaze Analytics. 
- Adds the missing events to handle Blaze campaign creation results. These are the main changes that should be reviewed https://github.com/woocommerce/woocommerce-android/pull/10926/commits/8c9f50d959c3cb474dcb5699381c2c0d4731be1e
- Adds back missing changes accidentally removed here: https://github.com/woocommerce/woocommerce-android/pull/10820/commits/ae11eefbff6a0b4734b59d2f1c2d6e309146114f#diff-9ca6a599d883aea20827d1df860489c7b8268aa2081c970308475898c27f8fa3 The changes related to reverting those accidental changes where already reviewed. So you can skip these commit changes: https://github.com/woocommerce/woocommerce-android/pull/10926/commits/73d47ba3a070f0a7cf19f244b58f04743226d414

### Testing instructions

1. Trigger Blaze campaign creation flow
2. Click on "Confirm details"
3. Click on "Submit Campaign"
4. Upon success, check the following is tracked: `🔵 Tracked: blaze_campaign_creation_success`

Repeat the steps, but before step 3, turn off the internet and check for the following event:
`blaze_campaign_creation_failed, Properties: {"blaze_creation_error":null` For the case of no connectivity there's no error message available so the value will be null. But we are mostly interested on API errors so this should not be an issue.
